### PR TITLE
Cypress/E2E: Fix e2e for sidebar channels

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix e2e for sidebar channels

### CODE REVIEW

File: e2e/cypress/integration/channel_sidebar/category_collapsing_spec.js [hunk 1]
```diff
@@ -128,7 +128,7 @@ describe('Channel sidebar', () => {
 
         // Wait for state to settle
         // eslint-disable-next-line cypress/no-unnecessary-waiting
-        cy.wait(TIMEOUTS.ONE_SEC);
+        cy.wait(TIMEOUTS.FIVE_SEC);
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel_sidebar/category_collapsing_spec.js:131]: Increase wait time to allow more time in saving of state to async storage.

File: e2e/cypress/integration/channel_sidebar/unread_filter_spec.js [hunk 2]
```diff
@@ -214,6 +224,7 @@ function disableUnreadFilter() {
 function createChannel(teamId, channelName, message) {
     cy.apiCreateChannel(teamId, channelName, channelName, 'O', '', '', false).then(({channel}) => {
         if (message) {
+            cy.wait(TIMEOUTS.THREE_SEC);
```
Comment (saturninoabril) [inline e2e/cypress/integration/channel_sidebar/unread_filter_spec.js:227]: This wait time is to have an interval between channel creation and posting message to the said channel, which guarantees receiving unread message.

**Comment by josephbaylon** [comment]
@saturninoabril For some reason, I couldn't get a successful run on both `dm_category_spec` and `unread_unfilter_spec`. The hang is consistent for me.

For `dm_category_spec`, it hangs on a previous test case which sets up the dm's needed by the updated test case
For `unread_unfilter_spec`, it hangs just towards the end

Note: I'm running against latest mattermost-server/mattermost-webapp/mattermost-redux/enterprise and did `make nuke && make test-data`. Not sure if it's a new server issue I'm experiencing.

Screen Shot 2021-01-28 at 9 34 19 AM
Screen Shot 2021-01-28 at 9 51 53 AM

**Comment by saturninoabril** [comment]
Maybe it's due to previous commit on server. I'm getting the following error locally with latest master on all repos.
```
store/sqlstore/store.go:271:4: db.SetConnMaxIdleTime undefined (type *sql.DB has no field or method SetConnMaxIdleTime)
```
If you're seeing the same, try to comment [this line](https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/store.go#L271) as it might be waiting for a PR or may need follow up change to fix it.

I've got all tests passed.
Screen Shot 2021-01-29 at 2 24 24 AM
Screen Shot 2021-01-29 at 2 23 05 AM

File: e2e/cypress/integration/channel_sidebar/category_collapsing_spec.js [hunk 3]
```diff
@@ -139,32 +139,38 @@ describe('Channel sidebar', () => {
...
+            cy.apiCreateChannel(teamId, 'channel-test', 'Channel Test').then(({channel}) => {
+                cy.postMessageAs({sender: sysadmin, message: 'Test', channelId: channel.id});
 
-        // * Verify that all categories are visible
-        cy.get('.SidebarChannelGroupHeader:contains(CHANNELS)').should('be.visible');
-        cy.get('.SidebarChannelGroupHeader:contains(CHANNELS) i').should('be.visible').should('not.have.class', 'icon-rotate-minus-90');
-        cy.get('.SidebarChannelGroupHeader:contains(DIRECT MESSAGES)').should('be.visible');
-        cy.get('.SidebarChannelGroupHeader:contains(DIRECT MESSAGES) i').should('be.visible').should('not.have.class', 'icon-rotate-minus-90');
+                // * Verify that CHANNELS starts expanded
+                cy.get('.SidebarChannelGroupHeader:contains(CHANNELS) i').should('not.have.class', 'icon-rotate-minus-90');
```
Comment (furqanmlk) [inline e2e/cypress/integration/channel_sidebar/category_collapsing_spec.js:147]: This ```cy.get('.SidebarChannelGroupHeader:contains``` is being used for 21 times in one file
can we make a local function with parameter

**Review by furqanmlk — APPROVED** [review]
All tests passed ✅ 

image

Just one question :)

**Review by josephbaylon — APPROVED** [review]
LGTM. Since they're passing for you guys locally, I'll approve PR.

**Comment by josephbaylon** [comment]
The db connection issue was due to go version. There was a discussion in developer channel [here](https://community.mattermost.com/core/pl/brmw7afiujro3f69rw65rozqqw).

**Pull request merged**